emcc.py

```diff
@@ -923,6 +930,15 @@ def detect_fixed_language_mode(args):
         final_suffix = 'eout' # not bitcode, not js; but just result from preprocessing stage of the input file
       if '-M' in newargs or '-MM' in newargs:
         final_suffix = 'mout' # not bitcode, not js; but just dependency rule of the input file
+
+      if final_suffix == None:
+        if target_output_type != None:
+          final_suffix = target_output_type
+        elif '.' in target:
+          final_suffix = target.split('.')[-1]
+        else:
+          final_suffix = ''
```

Comment:
For this case in particular, it would be good to add a comment about what this does. I presume it will build to an object file? (same as `.o`?)